integration/pull_test.go

```diff
@@ -845,3 +853,42 @@ func TestRpullImageThenRemove(t *testing.T) {
 
 	checkFuseMounts(t, sh, 0)
 }
+
+// TestRpullImageWithMinLayerSize pulls and rpulls an image with a runtime min_layer_size to confirm small layers are mounted locally
+func TestRpullImageWithMinLayerSize(t *testing.T) {
+	regConfig := newRegistryConfig()
+	sh, done := newShellWithRegistry(t, regConfig)
+	defer done()
+
+	// Setup environment
+	if err := testutil.WriteFileContents(sh, defaultContainerdConfigPath, getContainerdConfigYaml(t, false), 0600); err != nil {
+		t.Fatalf("failed to write %v: %v", defaultContainerdConfigPath, err)
+	}
+	if err := testutil.WriteFileContents(sh, defaultSnapshotterConfigPath, getSnapshotterConfigYaml(t, false, minLayerSizeConfig), 0600); err != nil {
+		t.Fatalf("failed to write %v: %v", defaultSnapshotterConfigPath, err)
+	}
+	sh.
+		X("update-ca-certificates").
+		Retry(100, "nerdctl", "login", "-u", regConfig.user, "-p", regConfig.pass, regConfig.host)
+
+	rebootContainerd(t, sh, "", "")
+
+	containerImage := rabbitmqImage
+
+	copyImage(sh, dockerhub(containerImage), regConfig.mirror(containerImage))
+	indexDigest := buildIndex(sh, regConfig.mirror(containerImage))
+
+	rawJSON := sh.O("soci", "index", "info", indexDigest)
+	var sociIndex soci.Index
+	if err := json.Unmarshal(rawJSON, &sociIndex); err != nil {
+		t.Fatalf("invalid soci index from digest %s: %v", indexDigest, rawJSON)
+	}
+
+	if len(sociIndex.Blobs) == 0 {
+		t.Fatalf("soci index %s contains 0 blobs, invalidating this test", indexDigest)
+	}
+
+	sh.X("soci", "image", "rpull", "--user", regConfig.creds(), "--soci-index-digest", indexDigest, regConfig.mirror(containerImage).ref)
+
+	checkFuseMounts(t, sh, 3) // default behavior produces 10 Fuse mounts
```

Comment:
Is this right? The default behavior should produce 3 mounts, since the defaultMinLayerSize is set to 10MiB.

Reply:
I did run it both ways, and it produced 10 without a runtime min layer size set. I'll see if the creation-time min layer size is getting overridden in some way, or what other explanation there is for my results.

Reply:
See other thread below.